test/10_erc1155_non_transferable.ts

```diff
@@ -0,0 +1,394 @@
+import {ethers} from 'hardhat';
+import {Signer, ContractFactory, Contract} from 'ethers';
+
+import {expect} from 'chai';
+import constants from '../testHelpers/constants';
+
+import Users from '../testHelpers/users';
+
+const BN = ethers.BigNumber.from;
+
+import {ERC1155NonTransferable} from '../typechain';
+
+let ERC1155NonTransferable_Factory: ContractFactory;
+
+import revertReasons from '../testHelpers/revertReasons';
+import {eventNames} from '../testHelpers/events';
+
+let users;
+
+describe('ERC1155 non transferable functionality', async () => {
+  before(async () => {
+    const signers: Signer[] = await ethers.getSigners();
+    users = new Users(signers);
+
+    ERC1155NonTransferable_Factory = await ethers.getContractFactory(
+      'ERC1155NonTransferable'
+    );
+  });
+
+  let contractERC1155NonTransferable: ERC1155NonTransferable;
+
+  async function deployContracts() {
+    contractERC1155NonTransferable = (await ERC1155NonTransferable_Factory.deploy(
+      '/non/transferable/uri'
```

Comment:
being a pedant, this isn't a uri, unless has a base somewhere and if so just shoot me...

Reply:
Changed to uri example for EIP1155 standard. Will be included in next commit.